### PR TITLE
refactor(form-field): signals everywhere !

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

Refactored `lu-form-field` to use signals in many places, fixing numerous change detection-related issues, including https://github.com/LuccaSA/lucca-front/issues/3233.

-----

This will require a quite big technical review to make sure we didn't break anything unwillingly, all my tests were fine but still.

-----

### CODE REVIEW

File: packages/ng/form-field/form-field.component.ts [hunk 1]
```diff
@@ -30,130 +31,87 @@ let nextId = 0;
...
 	 * Max amount of characters allowed, defaults to 0, which means hidden, no maximum
 	 */
-	@Input()
-	counter = 0;
-
-	required = false;
+	counter = input<number>(0);
 
 	get contentLength(): number {
 		return (this.#inputs[0]?.host?.nativeElement as HTMLInputElement)?.value.length || 0;
```
Comment (GuillaumeNury) [inline packages/ng/form-field/form-field.component.ts:117]: If `#inputs` becomes a `Signal`, we could have a `computed` for `contentLength` too

File: packages/ng/form-field/form-field.component.ts [hunk 2]
```diff
@@ -183,6 +141,20 @@ export class FormFieldComponent implements OnChanges, OnDestroy, DoCheck {
...
+			this.updateAria();
+		});
+
+		effect(() => {
+			this.#luClass.setState({
+				[`mod-${this.size()}`]: !!this.size(),
+				'mod-checkable': this.layout() === 'checkable',
+				'form-field': this.layout() !== 'fieldset',
+			});
+		});
```
Comment (GuillaumeNury) [inline packages/ng/form-field/form-field.component.ts:155]: this is satisfaying!

File: packages/ng/form-field/form-field.component.ts (hunk 1 above)
Comment (Supamiu) [reply packages/ng/form-field/form-field.component.ts:117]: I don't think we can make it a signal due to how it's built and updated

Comment (GuillaumeNury) [reply packages/ng/form-field/form-field.component.ts:117]: Indeed, we would have to listen to input/change event

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3246/storybook

**Pull request merged**